From 6ef7715763a9e5f0d3795785550ab2a81c3994c0 Mon Sep 17 00:00:00 2001
From: RagedMeteor1837 <148640131+RagedMeteor1837@users.noreply.github.com>
Date: Sun, 29 Mar 2026 21:37:30 +0100
Subject: [PATCH] now fix the update dialog too. its late


diff --git a/src/main/java/dev/figboot/olauncher/launcher/update/LauncherUpdateChecker.java b/src/main/java/dev/figboot/olauncher/launcher/update/LauncherUpdateChecker.java
index 1c93978..35f4fb9 100644
--- a/src/main/java/dev/figboot/olauncher/launcher/update/LauncherUpdateChecker.java
+++ b/src/main/java/dev/figboot/olauncher/launcher/update/LauncherUpdateChecker.java
@@ -36,7 +36,6 @@ public class LauncherUpdateChecker {
 
     private final Gson gson;
     private LauncherUpdateInfo info;
-    private boolean channelSwitchPending = false;
 
     public LauncherUpdateChecker() {
         gson = new GsonBuilder()
@@ -96,7 +95,6 @@ public class LauncherUpdateChecker {
         info.setLastCheck(Instant.MIN);
         info.setLatestRelease(null);
         info.getIgnoreVersions().clear();
-        channelSwitchPending = true;
         saveInfo();
         updateCheck();
     }
@@ -196,14 +194,10 @@ public class LauncherUpdateChecker {
                 new Object[]{info.getChannel(), release.getName(), release.getID()});
 
         if (info.getIgnoreVersions().contains(release.getID())) {
-            channelSwitchPending = false;
             return;
         }
 
-        boolean shouldUpdate = channelSwitchPending || shouldShowUpdateDialog(release);
-        channelSwitchPending = false;
-
-        if (!shouldUpdate) return;
+        if (!shouldShowUpdateDialog(release)) return;
 
         if (LauncherVersion.getInstance().isDirty()) {
             LOGGER.info("Update available ({}) but notification suppressed for dirty build.",
-- 
2.50.1 (Apple Git-155)

